bottlecap/src/metrics/enhanced/lambda.rs

```diff
@@ -343,6 +349,86 @@ impl Lambda {
         }
     }
 
+    pub fn generate_tmp_enhanced_metrics(
+        tmp_max: f64,
+        tmp_used: f64,
+        aggr: &mut std::sync::MutexGuard<Aggregator>,
+    ) {
+        let metric = Metric::new(
+            constants::TMP_MAX_METRIC.into(),
+            MetricValue::distribution(tmp_max),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("Failed to insert tmp_max metric: {}", e);
+        }
+
+        let metric = Metric::new(
+            constants::TMP_USED_METRIC.into(),
+            MetricValue::distribution(tmp_used),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("Failed to insert tmp_used metric: {}", e);
+        }
+
+        let tmp_free = tmp_max - tmp_used;
+        let metric = Metric::new(
+            constants::TMP_FREE_METRIC.into(),
+            MetricValue::distribution(tmp_free),
+            None,
+        );
+        if let Err(e) = aggr.insert(metric) {
+            error!("Failed to insert tmp_free metric: {}", e);
+        }
+    }
+
+    pub fn set_tmp_enhanced_metrics(&self, mut send_metrics: Receiver<()>) {
+        if !self.config.enhanced_metrics {
+            return;
+        }
+
+        let aggr = Arc::clone(&self.aggregator);
+
+        tokio::spawn(async move {
+            // Set tmp_max and initial value for tmp_used
+            let (bsize, blocks, bavail) = match statfs_info(constants::TMP_PATH) {
+                Ok(stats) => stats,
+                Err(err) => {
+                    debug!("Could not emit tmp enhanced metrics. {:?}", err);
+                    return;
+                }
+            };
+            let tmp_max = bsize * blocks;
+            let mut tmp_used = bsize * (blocks - bavail);
+
+            let mut interval = interval(Duration::from_millis(10));
```

Comment:
Why 10ms?

Reply:
This was the interval used in the Go agent so left it the same, a less frequent ticker may not capture the tmp data accurately